### ISSUE TITLE
Stuck at "2018-10-05 04:52:30: Saving images in: target"

### ISSUE THREAD

**Issue by nexisn4** [fperazzi/proSR#23]
I'm running
```
python test.py --checkpoint data/checkpoints/proSR.pth --target src/img1.png --output-dir target --scale 2 --cpu
```
and then getting the output
```
2018-10-05 04:52:30: phase: phase.TEST
2018-10-05 04:52:30: checkpoint: proSR.pth
{'G': {'bn_size': 4,
       'growth_rate': 40,
       'level_compression': -1,
       'level_config': [[8, 8, 8, 8, 8, 8, 8, 8, 8], [8, 8, 8], [8]],
       'max_num_feature': 312,
       'max_scale': 8,
       'num_init_features': 160,
       'ps_woReLU': False,
       'res_factor': 0.2,
       'residual_denseblock': True},
 'cmd': {'checkpoint': None,
         'config': None,
         'model': 'prosr',
         'no_curriculum': False,
         'output': 'data/checkpoints/proSR-flickr',
         'seed': 128,
         'visdom': True,
         'visdom_port': 2002},
 'data': {'input_size': [48, 36, 24], 'scale': [2, 4, 8]},
 'test': {'dataset': {'downscale': False,
                      'mean': [0.4488, 0.4371, 0.404],
                      'path': {'source': '', 'target': 'data/datasets/Set14'},
                      'stddev': [0.0039215, 0.0039215, 0.0039215]},
          'fast_validation': -1},
 'train': {'batch_size': 16,
           'dataset': {'downscale': False,
                       'mean': [0.4488, 0.4371, 0.404],
                       'path': {'source': '',
                                'target': 'data/datasets/Ensemble/**'},
                       'stddev': [0.0039215, 0.0039215, 0.0039215]},
           'epochs': 500,
           'growing_steps': [0.12, 0.25, 0.45, 0.6, 1.0],
           'io': {'eval_epoch_freq': 10,
                  'print_errors_freq': 100,
                  'save_model_freq': 20},
           'l1_loss_weight': 1.0,
           'lr': 0.0001,
           'lr_decay': 0.5,
           'lr_schedule_patience': 30,
           'smallest_lr': 1e-05}}
2018-10-05 04:52:30: Saving images in: target
```

However, it's stuck at this part. I've left it for hour or so and nothing happened other than CPU usage hovering 5-50% and RAM usage pretty much 100%. Any ideas?

**Comment by frankwang95** [comment]
I'm having similar issues.

**Comment by fperazzi** [comment]
I think this is pytorch related. The '--cpu' flag only set the tensor and
model to reside on the CPU. Do you experience the same problem running the
smaller model ProSRs.pth?

**Comment by nexisn4** [comment]
I tried ProSRs.pth and it seems to show the same error. It hangs and I'm forced to ctrl+c to force stop, showing the following:

```
^CProcess Process-9:
Process Process-15:
Process Process-14:
Process Process-8:
Process Process-5:
Process Process-4:
Process Process-16:
Process Process-3:
Process Process-12:
Process Process-7:
Process Process-6:
Process Process-10:
Process Process-11:
Process Process-13:
Process Process-2:
Process Process-1:
Traceback (most recent call last):
Traceback (most recent call last):
  File "/home/nexisn4/anaconda3/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/home/nexisn4/anaconda3/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
Traceback (most recent call last):
Traceback (most recent call last):
Traceback (most recent call last):
  File "/mnt/c/study/proSR/prosr/data/multiproc.py", line 45, in _worker_loop
    r = index_queue.get()
Traceback (most recent call last):
Traceback (most recent call last):
Traceback (most recent call last):
  File "/home/nexisn4/anaconda3/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
Traceback (most recent call last):
  File "/home/nexisn4/anaconda3/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
Traceback (most recent call last):
Traceback (most recent call last):
Traceback (most recent call last):
  File "/home/nexisn4/anaconda3/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/home/nexisn4/anaconda3/lib/python3.6/multiprocessing/queues.py", line 94, in get
    res = self._recv_bytes()
  File "/home/nexisn4/anaconda3/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
```
The error has a whole bunch more lines and ends with
```
  File "/home/nexisn4/anaconda3/lib/python3.6/site-packages/torch/nn/modules/container.py", line 91, in forward
    input = module(input)
  File "/home/nexisn4/anaconda3/lib/python3.6/site-packages/torch/nn/modules/module.py", line 491, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/nexisn4/anaconda3/lib/python3.6/site-packages/torch/nn/modules/conv.py", line 301, in forward
    self.padding, self.dilation, self.groups)
KeyboardInterrupt
```

It does indeed look like a pytorch problem, and I'm not sure if it's something that I can fix on my device...

**Comment by fperazzi** [comment]
Can you try setting`num_workers=0` in `DataLoader`?

**Comment by nexisn4** [comment]
It looks like the command works for small inputs. I gave a 18kb image and the command finished successfully in 8 seconds. I gave a bigger 5mb image and it did the same thing as before, and showed the same error when I killed it with ctrl+c. I let it run for a while and I saw a segmentation fault (core dumped) after 30-60 mins.

I tried modifying `num_workers` and `batch_size` but it looks like that's not having any affect on the output.
